### PR TITLE
[DOCS] Update pr template and remove enhancement feature type

### PR DESCRIPTION
**Pull request by OrtalG**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [ENHANCEMENT], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- update readme to have DOD checklist
- remove [ENHANCEMENT] as PR type 

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

Thank you for submitting!

### CODE REVIEW

**Comment by OrtalG** [comment]
@cla-bot check

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -18,4 +18,15 @@ Previous Design Review notes:
 -
 
 
-Thank you for submitting!
+### Definition of Done
+Please delete options that are not relevant.
+
+- [ ] My code follows the style guidelines of this project
```
Comment (abegong) [inline .github/pull_request_template.md]:
```suggestion
- [ ] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -18,4 +18,15 @@ Previous Design Review notes:
...
+- [ ] I have performed a self-review of my own code
```
Comment (abegong) [inline .github/pull_request_template.md]:
```suggestion
- [ ] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -18,4 +18,15 @@ Previous Design Review notes:
...
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] I have added unit tests where applicable and made sure that new and existing tests are passing.
```
Comment (abegong) [inline .github/pull_request_template.md]:
```suggestion
- [ ] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
```

**Review by abegong — APPROVED** [review]
This makes sense to me. I've added some suggestions for links within the checklist, but I haven't fully harmonized those docs.

**Pull request merged**